src/Grpc.Net.Client/Internal/OperatingSystem.cs

```diff
@@ -41,5 +45,7 @@ private OperatingSystem()
 #else
         IsAndroid = false;
 #endif
+        IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        OSVersion = Environment.OSVersion.Version;
```

Comment:
excerpt from offline discussion: 

> I experimented with Environment.OSVersion and one issue I found is that when called from .NET Framework, it seems to return garbage on newer windows versions (e.g. I tried on Microsoft Windows Server 2019 Datacenter, 10.0.17763 Build 17763 and it returned "Microsoft Windows NT 6.2.9200.0"). Looks like the value returned is the latest windows version the .NET Framework knows about (Which is way too old now and not useful to us).
So some better way of detecting OS Version will be needed.

Can you confirm that this has been taken into account and that we really rely on Environment.OSVersion.Version providing the right info when on .NET Framework (which is actually where we care about it the most)?
This would also deserve a test that actually targets net462 (not sure if we currently have one).

Reply:
Btw, if Environment.OSVersion.Version gives outdated info about build number, in practice this would mean that a high-enough build number would never be detected and creating the GrpcChannel would always throw on .NET Framework?